examples/search_examples/server/routes/server_search_route.ts

```diff
@@ -26,36 +27,41 @@ export function registerServerSearchRoute(router: IRouter<DataRequestHandlerCont
     },
     async (context, request, response) => {
       const { index, field } = request.query;
-      // Run a synchronous search server side, by enforcing a high keepalive and waiting for completion.
-      // If you wish to run the search with polling (in basic+), you'd have to poll on the search API.
-      // Please reach out to the @app-arch-team if you need this to be implemented.
-      const res = await context
-        .search!.search(
-          {
-            params: {
-              index,
-              body: {
-                aggs: {
-                  '1': {
-                    avg: {
-                      field,
+
+      const abortController = new AbortController();
+      request.events.aborted$.subscribe(() => {
```

Comment:
this is a new thing that we didn't do before but should encourage everywhere where we run possible long-running searches server-side. 

Without this, it might happen that user has already terminated the original request, but search still running polling for results.

see: https://github.com/elastic/kibana/issues/113423

Reply:
Do you mind adding a comment in code here essentially describing what you've mentioned here?

Also, we have a little helper function that simplifies things a tiny bit here: https://github.com/elastic/kibana/blob/66aca8b486342727de3f5c95821bbeb22fae3d29/src/plugins/data/server/lib/get_request_aborted_signal.ts

As I'm looking at this, I'm wondering... Will these subscriptions be properly GC'd? It brings back to mind the memory leak we encountered at some point.

Reply:
> As I'm looking at this, I'm wondering... Will these subscriptions be properly GC'd? It brings back to mind the memory leak we encountered at some point.

Yes, this is fine, because `events.aborted$` completes when the request completes

Reply:
> Also, we have a little helper function that simplifies things a tiny bit here: 
> https://github.com/elastic/kibana/blob/66aca8b486342727de3f5c95821bbeb22fae3d29/src/plugins/data/server/lib/get_request_aborted_signal.ts

Realised that server of example plugins can't import static code from othe plugins